### PR TITLE
Fix server error banned login

### LINKED ISSUE #253: Better message when suspended from logging in
When a login suspended user tries to log in they get `Internal Server Error` message. This should probably be improved to show something more informative, possible display a reason for suspension instead.

### PR DESCRIPTION
**Pull request by geekingfrog**
Fix #253 (and #301 which is a duplicate).

### CODE REVIEW

File: lib/teiserver/account/libs/user_lib.ex [hunk 1]
```diff
@@ -351,4 +357,17 @@ defmodule Teiserver.Account.UserLib do
...
     end)
   end
+
+  defp can_login(user) do
+    cond do
+      Teiserver.CacheUser.is_restricted?(user.id, ["Login"]) ->
+        {:error, "You are banned"}
+
+      user.smurf_of_id != nil ->
+        {:error, "Smurf detected"}
```
Comment (StanczakDominik) [inline lib/teiserver/account/libs/user_lib.ex:367]:
```suggestion
        {:error, "Alt account detected. Please log in using your original account instead. If you're not sure what that account is or have trouble accessing it, please contact the moderation team at https://discord.gg/beyondallreason -> #open-ticket"}
```

File: lib/teiserver/account/libs/user_lib.ex [hunk 2]
```diff
@@ -351,4 +357,17 @@ defmodule Teiserver.Account.UserLib do
       {key, Teiserver.store_get(:restriction_lookup_store, key)}
     end)
   end
+
+  defp can_login(user) do
+    cond do
+      Teiserver.CacheUser.is_restricted?(user.id, ["Login"]) ->
+        {:error, "You are banned"}
```
Comment (StanczakDominik) [inline lib/teiserver/account/libs/user_lib.ex:364]:
```suggestion
        {:error, "Your account is currently suspended. Check the suspension's status at https://discord.gg/beyondallreason -> #moderation-bot"}
```

**Review by StanczakDominik — APPROVED** [review]
I've deployed it on server5, tested, and it *just works* 👌🏻 this is amazing, thank you! That's one less annoying copypasta we have to carry around 😁 

If you find those proposed messages clear and understandable, we can just merge it right away :)

**Comment by geekingfrog** [comment]
Cool, ship it \o/

**Pull request merged**